packages/pretty-format/src/plugins/lib/markup.ts

```diff
@@ -62,14 +62,23 @@ export const printChildren = (
   printer: Printer,
 ): string =>
   children
-    .map(
-      child =>
-        config.spacingOuter +
-        indentation +
-        (typeof child === 'string'
+    .map(child => {
+      const printedChild =
+        typeof child === 'string'
           ? printText(child, config)
-          : printer(child, config, indentation, depth, refs)),
-    )
+          : printer(child, config, indentation, depth, refs);
+
+      if (
+        printedChild === '' &&
+        typeof child === 'object' &&
+        child !== null &&
+        (child as Node).nodeType !== 3
```

Comment:
That's not pretty but required to not break existing tests. Personally, I would like to filter any empty strings. Maybe add an option "ignoreEmptyString" and use that instead?

Reply:
what's `nodeType: 3`? some comment node? (worth an inline code comment I'd say)